boostsec/registry_validator/upload_rules_db.py

```diff
@@ -0,0 +1,134 @@
+"""Uploads the Rules DB file."""
+import argparse
+import json
+import sys
+from pathlib import Path
+from typing import Any
+from urllib.parse import urljoin
+
+import requests
+import yaml
+
+MUTATION = """
+mutation setRules($rules: RuleInputSchemas!) {
+    setRules(namespacedRules: $rules){
+        __typename
+        ... on RuleSuccessSchema {
+            successMessage
+        }
+        ... on RuleErrorSchema {
+            errorMessage
+        }
+    }
+}"""
+
+
+def _log_error_and_exit(message: str) -> None:
+    """Log an error message and exit."""
+    print("ERROR: " + message)
+    sys.exit(1)
+
+
+def find_modules(modules_path: str) -> list[Path]:
```

Comment:
reading this top to bottom, it was surprised to see .splitlines() on a "modules_path"
purely for readability, might be worth changing this to a list of strings and splitting it in `main`